docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+# cuDF Architecture
+
+The cuDF library is a GPU-accelerated, [Pandas-like](https://pandas.pydata.org/) DataFrame library.
+pandas APIs provide users a greate deal of power and flexibility, and we aim to match that.
+As a result, a key design challenge for cuDF is finding the simplest, most performant approaches to mimic pandas APIs.
+
+At a high level, cuDF is structured in three layers, each of which serves a distinct purpose in this regard:
+
+1. The Frame layer: The user-facing implementation of pandas-like data structures.
+2. The Column layer: The core internal data structures used to bridge the gap to our lower-level implementations.
+3. The Cython layer: The wrappers around the fast C++ `libcudf` library.
+
+In this document we will review each of these layers, their roles, and the requisite tradeoffs.
+
+
+## The Frame layer
```

Comment:
It takes a bit of reading in this section before we unravel that the `Frame` layer is a collection of classes, mostly connected by inheritance, that ultimately provides the user-facing `Series`, `DataFrame` and `Index` types.

Should we include something like an (ascii) tree that shows all the types in the Frame layer and their relationship? Or is that too simple a view of things?

Reply:
I think that's a great idea. I actually considered this but abandoned it as over the top. I'll revisit.